pkg/controller/serviceaccount/tokens_controller.go

```diff
@@ -620,3 +711,37 @@ func parseSecretQueueKey(key interface{}) (secretQueueKey, error) {
 func makeCacheKey(namespace, name string) string {
 	return namespace + "/" + name
 }
+
+func podSecretIndexFunc() func(obj interface{}) ([]string, error) {
+	return func(obj interface{}) ([]string, error) {
+		pod, ok := obj.(*v1.Pod)
+		if !ok {
+			return []string{}, nil
+		}
+		keys := []string{}
+		for _, volume := range pod.Spec.Volumes {
+			if vs := volume.VolumeSource.Secret; vs != nil {
+				secretName := vs.SecretName
+				keys = append(keys, fmt.Sprintf("%s/%s", pod.Namespace, secretName))
+			}
+		}
```

Comment:
whether we do this on-demand or in an indexer, use the VisitPodSecretNames helper to determine what secrets the pod references ... there's lots of ways a pod can grab a value from a secret other than a volume mount